ql/src/java/org/apache/hadoop/hive/ql/reexec/ReExecDriver.java

```diff
@@ -148,8 +148,7 @@ public void setOperationId(String operationId) {
   @Override
   public CommandProcessorResponse run() throws CommandProcessorException {
     executionIndex = 0;
-    int maxExecutuions = 1 + coreDriver.getConf().getIntVar(ConfVars.HIVE_QUERY_MAX_REEXECUTION_COUNT);
-
+    int maxExecutions = getMaxExecutions();
```

Comment:
Is this still needed in `ReExecDriver`? Shouldn't it be driven completely by the plugins implementation now, i.e., `shouldReExecute` will return false after the number of retries exceeds the max?

Reply:
I would like to avoid, that some new plugin forgets to check the max in its shouldReExecute and we go in an infinite loop.

Reply:
Makes sense. Can we leave a comment mentioning that? Thanks

Reply:
I also do not like this approach as you are aggregating all the conditions from underlying plugins. What you could do is to define default shouldReExecute method under IReExecutionPlugin, in this case new plugin would use that or has to override it.

Reply:
That would not solve the problem, every plugin will override the shouldReExecute method, that is the main goal of a plugin. Still they can ignore the max execution property. This is here for a failsafe, to not have infinite loops.

Reply:
could we stick in this case with some hardcoded limit?

Reply:
in getMaxReExecutions could we then just stick with some default hardcoded value?